### PR TITLE
Update modules

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
TODO:

- new dragmap version causes Segmentation fault
- [x] Run Mutect2 manual tests

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Can you add such a section with the newly updated tools:
https://github.com/nf-core/sarek/blob/dev/CHANGELOG.md#dependencies

File: workflows/sarek.nf [hunk 1]
```diff
@@ -436,7 +436,10 @@ workflow SAREK {
 
             save_trimmed_fail = false
             save_merged = false
-            FASTP(ch_reads_fastp, save_trimmed_fail, save_merged)
+            FASTP(ch_reads_fastp,
+                    [], // we are not using any dapater fastas at the moment
```
Comment (maxulysse) [inline workflows/sarek.nf:440]:
```suggestion
                    [], // we are not using any adapter fastas at the moment
```

Comment (FriederikeHanssen) [reply workflows/sarek.nf:440]: 🤣

**Comment by FriederikeHanssen** [comment]
```
Run nf-core/setup-nextflow@v1
Error: Could not retrieve Nextflow release matching 21.10.3.
API rate limit exceeded for installation ID 1565271.
Error: Could not parse the download URL
Cannot read properties of undefined (reading 'filter')
Error: versionSpec parameter is required
Warning: Nextflow appears to have installed correctly, but an error was thrown while running it.
```

that time of the week again

File: CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +48,43 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+| `cnvkit`               | 0.9.9       | unchanged   |
+| `control-freec`        | 11.6        | unchanged   |
+| `deepvariant`          | 1.3.0       | unchanged   |
+| `dragmap`              | 1.2.1       | unchanged   |
+| `ensembl-vep`          | 106.1       | unchanged   | //should we update this here? |
+| `fastp`                | 0.23.2      | unchanged   |
+| `fastqc`               | 0.11.9      | unchanged   |
+| `fgbio`                | 2.0.2       | unchanged   |
+| `freebayes`            | 1.3.5       | unchanged   |
+| `gatk4`                | 4.2.6.1     | 4.3.0.0     |
```
Comment (maxulysse) [inline CHANGELOG.md:69]: I think we can keep only the updated ones

File: CHANGELOG.md [hunk 3]
```diff
@@ -47,6 +48,43 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+| `ascat`                | 3.0.0       | unchanged   |
+| `bcftools`             | 1.15.1      | unchanged   |
+| `bwa-mem2`             | 2.2.1       | unchanged   |
+| `bwa`                  | 0.7.17      | unchanged   |
+| `cancerit-allelecount` | 4.3.0       | unchanged   |
+| `cnvkit`               | 0.9.9       | unchanged   |
+| `control-freec`        | 11.6        | unchanged   |
+| `deepvariant`          | 1.3.0       | unchanged   |
+| `dragmap`              | 1.2.1       | unchanged   |
+| `ensembl-vep`          | 106.1       | unchanged   | //should we update this here? |
```
Comment (maxulysse) [inline CHANGELOG.md:64]: Did I forgot to update that previously?

File: CHANGELOG.md (hunk 2 above)
Comment (FriederikeHanssen) [reply CHANGELOG.md:69]: more to come, see modules repo 😆

File: CHANGELOG.md (hunk 3 above)
Comment (FriederikeHanssen) [reply CHANGELOG.md:64]: think so, do you have time to update vep, I think they are at 108, or should we push it? (you should show me how I can do that myself :) )

Comment (maxulysse) [reply CHANGELOG.md:64]: Updating module is easy, just the one thing we need to do on top of that is building the containers with cache

File: modules/local/sambamba.nf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+//               Any parameters that need to be evaluated in the context of a particular sample
+//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
+// TODO nf-core: Software that can be piped together SHOULD be added to separate module files
+//               unless there is a run-time, storage advantage in implementing in this way
+//               e.g. it's ok to have a single module for bwa to output BAM instead of SAM:
+//                 bwa mem | samtools view -B -T ref.fasta
+// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
+//               list (`[]`) instead of a file can be used to work around this issue.
+
+process SAMBAMBA {
```
Comment (maxulysse) [inline modules/local/sambamba.nf:18]: we don't have that in nf-core/modules yet?

**Review by maxulysse — COMMENTED** [review]
Looking good

File: modules/local/sambamba.nf (hunk 4 above)
Comment (FriederikeHanssen) [reply modules/local/sambamba.nf:18]: 🤦‍♀️ accidental commit, this was more benchmarking if smabamba is faster than gatk MD (it wasn't)

**Comment by FriederikeHanssen** [comment]
not sure whats up with all the test failures picked some random ones to run locally and they worked fine.

**Comment by FriederikeHanssen** [comment]
ok think I found all to update, only one missing is svdb (some issues with getting the tests to pass with the new version) and VEP potentially

File: tests/csv/3.0/tumor_mutational_burden.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+patient,sample,vcf,tbi,variantcaller,annotation
+hcc1395,hcc1395,/Users/monarchy/Projects/Coding/sarek/HCC1395T_vs_HCC1395N.mutect2.filtered_VEP.ann.vcf.gz,/Users/monarchy/Projects/Coding/sarek/HCC1395T_vs_HCC1395N.mutect2.filtered_VEP.ann.vcf.gz.tbi,mutect2,vep
```
Comment (maxulysse) [inline tests/csv/3.0/tumor_mutational_burden.csv:2]: Hardcoded?

Comment (FriederikeHanssen) [reply tests/csv/3.0/tumor_mutational_burden.csv:2]: ah yes need to remove it again from this. Couldn't test the tmb stuff yet with the small test data but needed actual files.

**Comment by FriederikeHanssen** [comment]
Ran mutect2 manual tests as well, look fine

File: modules/nf-core/dragmap/align/dragmap-align.diff [hunk 6]
```diff
@@ -0,0 +1,19 @@
+Changes in module 'nf-core/dragmap/align'
+--- modules/nf-core/dragmap/align/main.nf
++++ modules/nf-core/dragmap/align/main.nf
+@@ -2,10 +2,10 @@
+     tag "$meta.id"
+     label 'process_high'
+ 
+-    conda (params.enable_conda ? "bioconda::dragmap=1.3.0 bioconda::samtools=1.16.1 conda-forge::pigz=2.3.4" : null)
++    conda (params.enable_conda ? "bioconda::dragmap=1.2.1 bioconda::samtools=1.15.1 conda-forge::pigz=2.3.4" : null)
```
Comment (maxulysse) [inline modules/nf-core/dragmap/align/dragmap-align.diff:9]: why still using the old container?

Comment (FriederikeHanssen) [reply modules/nf-core/dragmap/align/dragmap-align.diff:9]: new version has a segmentation fault we can|t figure out at the moment. @Emiller88 saw the same in nascent

Comment (FriederikeHanssen) [reply modules/nf-core/dragmap/align/dragmap-align.diff:9]: I have opened an issue in the illumina repo, let's see what they say

**Review by maxulysse — APPROVED** [review]

**Pull request merged**